### PR TITLE
Add telemetry to S3 browser in preperation for merging to master

### PR DESCRIPTION
**Pull request by hunterwerlla**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/ProgressOutputStream.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class ProgressOutputStream(private val outputStream: OutputStream, val size: Lon
     }
 
     override fun write(b: ByteArray?, off: Int, len: Int) {
-        outputStream.write(b, off, len)
+        b?.let { outputStream.write(b, off, len) }
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/ProgressOutputStream.kt]: Should we allow null?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt [hunk 2]
```diff
@@ -65,6 +65,15 @@ interface TelemetryService : Disposable {
             ApplicationManager.getApplication().messageBus.connect().subscribe(TELEMETRY_TOPIC, notifier)
         }
 
+        @JvmStatic
+        fun recordSimpleTelemetry(project: Project?, name: String, result: TelemetryConstants.TelemetryResult, count: Double = 1.0) =
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt]: I like the `recordSimpleTelemetry` method. We seem to use a ternary type style a few times, does it make sense to have an overload that takes a `success` `Boolean` (or function that returns a `Boolean` as well?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/ProgressOutputStream.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/ProgressOutputStream.kt]: This is actually annotated `@notnull`, so changed to not allow null

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt]: Sounds like something we can used in a few places, added

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt [hunk 3]
```diff
@@ -67,4 +76,8 @@ class CreateS3BucketDialog(
 
     @TestOnly
     fun validateBucketName(): String? = if (bucketName().isEmpty()) message("s3.create.bucket.missing.bucket.name") else null
+
+    companion object {
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: make private

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt [hunk 4]
```diff
@@ -67,4 +73,8 @@ class DeleteObjectAction(
             .build()
         client.deleteObjects(deleteObjectsRequest)
     }
+
+    companion object {
+        const val TELEMETRY_NAME = "s3_deleteobject"
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt]: make private

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt [hunk 5]
```diff
@@ -88,4 +99,8 @@ class DownloadObjectAction(
             }
         })
     }
+
+    companion object {
+        const val TELEMETRY_NAME = "s3_downloadobject"
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt]: private private

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt]: Fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt (hunk 4 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DeleteObjectAction.kt]: fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt (hunk 3 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: fixed

**Pull request merged**